hatchet/graphframe.py

```diff
@@ -76,6 +76,134 @@ def __init__(self, graph, dataframe, exc_metrics=None, inc_metrics=None):
         self.exc_metrics = [] if exc_metrics is None else exc_metrics
         self.inc_metrics = [] if inc_metrics is None else inc_metrics
 
+    @staticmethod
+    def from_path(dirname_or_data, query, profile_format=None):
+        """Read a database directory and automatically detect the data format.
+
+        Arguments:
+            dirname_or_data (str): path to the database directory  
+            query (str): 
+            profile_format (str): Override for the profile_format, if detection
+            of format fails. 
+        """
+        FROM_DATABASE_MAPPER = {
+            'hpctoolkit': GraphFrame.from_hpctoolkit(dirname_or_data),
+            'caliper': GraphFrame.from_caliper(dirname_or_data, query),
+            'caliper_json': GraphFrame.from_caliper_json(dirname_or_data),
+            'gprof_dot': GraphFrame.from_gprof_dot(dirname_or_data),
+            'cprofile': GraphFrame.from_cprofile(dirname_or_data),
+            'pyinstrument': GraphFrame.from_pyinstrument(dirname_or_data),
+            'timemory': GraphFrame.from_timemory(dirname_or_data),
+            'literal': GraphFrame.from_literal(dirname_or_data),
+            'lists': GraphFrame.from_lists(dirname_or_data),
+        }
+
+        if profile_format:
```

Comment:
`if profile_format is not None`

note that your condition will work, but i dont like using it. by explicitly writing `is not None` you tell the rreader what you want and also catch the case wherre i pass profilee_format=True (a bool)